Should tslib be a production dependency?

<!--
  ⚠️⚠️ Please do the following before submitting: ⚠️⚠️

  📖 Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/
  🔎 Please search existing issues to avoid creating duplicates.
-->

### Describe the bug

When I try to import VSCodeButton (`import { VSCodeButton } from '@vscode/webview-ui-toolkit/react';`), I'm getting the following error:

```
    Cannot find module 'tslib' from '../../node_modules/.pnpm/@vscode+webview-ui-toolkit@1.2.1_react@18.2.0/node_modules/@vscode/webview-ui-toolkit/dist/button/index.js'

    Require stack:
..../node_modules/.pnpm/@vscode+webview-ui-toolkit@1.2.1_react@18.2.0/node_modules/@vscode/webview-ui-toolkit/dist/button/index.js
..../node_modules/.pnpm/@vscode+webview-ui-toolkit@1.2.1_react@18.2.0/node_modules/@vscode/webview-ui-toolkit/dist/custom-elements.js
..../node_modules/.pnpm/@vscode+webview-ui-toolkit@1.2.1_react@18.2.0/node_modules/@vscode/webview-ui-toolkit/dist/index.js
..../node_modules/.pnpm/@vscode+webview-ui-toolkit@1.2.1_react@18.2.0/node_modules/@vscode/webview-ui-toolkit/react/index.js
...........
```

In `./dist/button/index.js` there is an import of tslib (`import { __decorate } from "tslib";`), but tslib is defined as dev dependency of vscode-webview-ui-toolkit.

<!-- A clear and concise description of what the bug is. -->

### To reproduce

Run the import as above

<!-- Steps to reproduce the behavior. -->

### Desktop (please complete the following information):

-   OS Version: macOS 13.1
-   Toolkit Version: 1.2.1

I'm using pnpm version 7.5.2

Hi @hawkticehurst 

Sure, you can reproduce it with these steps (I'm using node 18.x but I guess any would do):
```
corepack enable
corepack prepare yarn@stable --activate
yarn dlx create-react-app --template typescript vscode-webview-ui-tooklit-tslib-issue
cd vscode-webview-ui-tooklit-tslib-issue
rm src/App.test.tsx
yarn add @vscode/webview-ui-toolkit
```
Replace src/App.tsx with:
```tsx
import React from 'react';
import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";

function App() {
  return (
    <VSCodeButton/>
  );
}

export default App;
```
and then `yarn start` should fail with these errors:
```
Module not found: Error: Can't resolve 'tslib' in '/home/florent/vscode-webview-ui-tooklit-tslib-issue/.yarn/__virtual__/@vscode-webview-ui-toolkit-virtual-8baf9fd80e/0/cache/@vscode-webview-ui-toolkit-npm-1.2.2-243130d45d-fd41c59310.zip/node_modules/@vscode/webview-ui-toolkit/dist/button'
ERROR in ./.yarn/__virtual__/@vscode-webview-ui-toolkit-virtual-8baf9fd80e/0/cache/@vscode-webview-ui-toolkit-npm-1.2.2-243130d45d-fd41c59310.zip/node_modules/@vscode/webview-ui-toolkit/dist/button/index.js 3:0-35
Module not found: Error: Can't resolve 'tslib' in '/home/florent/vscode-webview-ui-tooklit-tslib-issue/.yarn/__virtual__/@vscode-webview-ui-toolkit-virtual-8baf9fd80e/0/cache/@vscode-webview-ui-toolkit-npm-1.2.2-243130d45d-fd41c59310.zip/node_modules/@vscode/webview-ui-toolkit/dist/button'

ERROR in ./.yarn/__virtual__/@vscode-webview-ui-toolkit-virtual-8baf9fd80e/0/cache/@vscode-webview-ui-toolkit-npm-1.2.2-243130d45d-fd41c59310.zip/node_modules/@vscode/webview-ui-toolkit/dist/radio-group/index.js 3:0-85
Module not found: Error: Can't resolve '@microsoft/fast-web-utilities' in '/home/florent/vscode-webview-ui-tooklit-tslib-issue/.yarn/__virtual__/@vscode-webview-ui-toolkit-virtual-8baf9fd80e/0/cache/@vscode-webview-ui-toolkit-npm-1.2.2-243130d45d-fd41c59310.zip/node_modules/@vscode/webview-ui-toolkit/dist/radio-group'
```
As a bonus you also get the issue with the missing "@microsoft/fast-web-utilities" that I listed in my previous message.

NB: with this template, you also get an error from eslint, but it's not related.

Hope this helps.

--- a/package.json
+++ b/package.json
@@ -31,7 +31,8 @@
 	"dependencies": {
 		"@microsoft/fast-element": "^1.12.0",
 		"@microsoft/fast-foundation": "^2.49.4",
-		"@microsoft/fast-react-wrapper": "^0.3.22"
+		"@microsoft/fast-react-wrapper": "^0.3.22",
+		"tslib": "^2.6.2"
 	},
 	"peerDependencies": {
 		"react": ">=16.9.0"
@@ -53,7 +54,6 @@
 		"rollup-plugin-filesize": "^9.1.1",
 		"rollup-plugin-terser": "^7.0.2",
 		"rollup-plugin-transform-tagged-template": "0.0.3",
-		"tslib": "^2.1.0",
 		"typescript": "^4.6.2"
 	},
 	"keywords": [

